### PR TITLE
Add window_title addon

### PR DESCRIPTION
**Pull request by cairface**
Sets the window title based on the character you're currently logged in to

### CODE REVIEW

File: window_title/window_title.lua [hunk 1]
```diff
@@ -0,0 +1,63 @@
+local command = require('command')
```
Comment (cairface) [inline window_title/window_title.lua]: whoops

**Comment by alphaONE2** [comment]
Damn it! Why am I so blind? I need to step up my game.

**Comment by Nifim** [comment]
critical error while multi boxing 

image from console of crash
https://media.discordapp.net/attachments/338605882185351169/463180803115974656/unknown.png?width=700&height=480

File: window_title/window_title.lua [hunk 2]
```diff
@@ -0,0 +1,59 @@
+local account = require('account')
+local ffi = require('ffi')
+
+ffi.cdef[[
+  bool SetWindowTextW(void* hWnd, wchar_t const* title);
```
Comment (z16) [inline window_title/window_title.lua]: Two spaces? o.o

File: window_title/window_title.lua [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  void* GetActiveWindow();
+  ]]
```
Comment (z16) [inline window_title/window_title.lua]: This should not be indented.

File: window_title/window_title.lua [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  
+local user32 = ffi.load('user32')
+
+local wide = function(str)
```
Comment (z16) [inline window_title/window_title.lua]: We should provide a mechanism to do this, so I would mark this as temporary. Could be made part of #43.

File: window_title/window_title.lua [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+  local ctor = ffi.typeof('wchar_t[?]')
+  local wstr = ctor(#str + 1)
+  for i = 1, #str do
+    wstr[i - 1] = str:byte(i)
+  end
+  return wstr
+end
+
+local set_window_title = function(title)
+  local hwnd = user32.GetActiveWindow()
```
Comment (z16) [inline window_title/window_title.lua]: Will this work when dual boxing?

File: window_title/window_title.lua [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+end
+
+local set_window_title = function(title)
+  local hwnd = user32.GetActiveWindow() 
+  assert(hwnd, 'Could not obtain the current window handle.')
+  assert(type(title) == 'string', 'The window title must be a string.')
+  assert(user32.SetWindowTextW(hwnd, wide(title)), 'Critical error occurred renaming the window.')
+end
+
+do
```
Comment (z16) [inline window_title/window_title.lua]: Don't see the need for this `do` block.

File: window_title/window_title.lua (hunk 5 above)
Comment (cairface) [reply window_title/window_title.lua]: Turns out no! 🗡

File: window_title/window_title.lua (hunk 2 above)
Comment (cairface) [reply window_title/window_title.lua]: I dunno :(

**Comment by cairface** [comment]
So, we either need to expose the game's window handle from core or I could find it a bit haphazardly through ffi.

**Comment by Nifim** [comment]
Worked on first box i loaded, second box ended up blank 🤔

**Pull request merged**